gammapy/cube/fit.py

```diff
@@ -64,6 +64,11 @@ def __init__(
             model=self.model, exposure=exposure, psf=self.psf, edisp=self.edisp
         )
 
+    @property
+    def data_shape(self):
+        """Shape of the counts data"""
+        return self.counts.data.shape
```

Comment:
I think this is fine for the moment.
Ultimately, it is probably better to compare the `MapGeom` because that is what needs to be identical. The issue being that a `FluxPointDataset` or a `SpectrumDataset` doesn't have a `Geom`.

Reply:
Shouldn't there be a test specifically for `MapDataset` testing the properties as well as the likelihood methods?

Reply:
Indeed I'll add a test for `MapDataset`...